runner/clients/proxy/proxy.go

```diff
@@ -146,24 +161,103 @@ func (p *ProxyServer) handleRequest(w http.ResponseWriter, r *http.Request) {
 	}
 	w.WriteHeader(resp.StatusCode)
 
-	_, err = io.Copy(w, resp.Body)
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		p.log.Error("Error reading response body", "err", err)
+		return
+	}
+
+	_, err = w.Write(respBody)
 	if err != nil {
 		p.log.Error("Error copying response body", "err", err)
 	}
+
+	// p.DebugResponse(respBody)
 }
 
-func (p *ProxyServer) OverrideRequest(method string, params json.RawMessage) (bool, json.RawMessage, error) {
+func (p *ProxyServer) OverrideRequest(method string, rawParams json.RawMessage) (bool, json.RawMessage, error) {
+	p.log.Debug("intercepting request", "method", method)
+
 	switch method {
-	// Example of how to intercept a request
-	// case "eth_getBlockByNumber":
-	// 	response := "0x100"
-	// 	return true, json.RawMessage(`"` + response + `"`), nil
+	case "eth_getTransactionCount":
+		hexNonce := fmt.Sprintf("0x%x", p.nonce)
+		jsonResponse, _ := json.Marshal(hexNonce)
+		return true, jsonResponse, nil
```

Comment:
I think we should only override this request if we have a transaction from the same sender as requested. If we don't have a transaction in our mempool, just pass it through to the normal backend.